### ISSUE TITLE
Interactive viewers should be able to run queries in the Explore view with custom dimensions

### ISSUE THREAD

**Issue by rephus** [lightdash/lightdash#12045]
### Description

Right now, interactive viewers can't explore from here on charts with custom dimensions, because they are only available for developers. We need to find a way to allow `run those custom dimensions` without modifying the underlying SQL.

How I think this might work
- we link the saved chart in the explore in a query param (we do already things like that for dashboard_uuid or context) , when doing explore from here from a dashboard
- when making calls to /results using metricQuery, we pass the savedChartUuid to the backend
- the backend then checks if there are custom dimensions, if so, instead of throwing an error, it will copy the sql from the saved chart, rather than the metric query
This logic is similar to what we do with dashboard filters

Thread: https://lightdash.slack.com/archives/C04E1CCKN64/p1729068273012209?thread_ts=1728989933.197199&cid=C04E1CCKN64

**Comment by ZeRego** [comment]
@rephus what happens when they save that query/chart? Saving a reference to the original chart would create a web of delicate dependencies between charts. Once the custom dimension is deleted from the first one, all the other charts would throw an error.

My idea was to stop storing the custom dimensions at a query/chart level and make them reusable. Developers could create a virtual dimension for a model that is accessible to everyone. Similar to virtual views.

**Comment by rephus** [comment]
> what happens when they save that query/chart?

When saving a chart, we in the backend can copy the `custom dimension` SQL and add it to the new saved chart , there is no need to keep a reference or dependencies. The point is to rely on the infromation that is already stored, without allowing the user to modify the SQL on the metric query. 

> My idea was to stop storing the custom dimensions at a query/chart level and make them reusable (users have asked for this). Developers could create a virtual dimension for a model that is accessible to everyone. Similar to virtual views.

That is also another good approach :+1: It will require more work and some migrations, but it might be worth it if we can unlock other features like `sharing custom dimensions on charts`

**Comment by ZeRego** [comment]
tldr:

Javi approach:
- faster but more delicate, prone to security flows
- worse for governance as it would be easier to duplicate custom sql and no way to fix it across all charts

My approach:
- needs more thought and time to implement
- easier to govern as all custom dims would be centralized and reusable. But this might not be want all users want. When I asked for feedback on custom SQL dimension, users wanted both, reusable and non reusable.

**Comment by jpetey75** [comment]
Closing as shipped 🎉 Interactive viewers and editors can now run, refresh, filter, drill into and export any query or chart containing custom dimensions — the "cannot run queries with custom SQL dimensions" error was removed in [#22541](https://github.com/lightdash/lightdash/pull/22541).

Authoring SQL custom dimensions still requires the "Manage Custom Fields" permission — developer+ by default, but grantable to editors/interactive viewers via custom roles ([#22082](https://github.com/lightdash/lightdash/pull/22082)). Bin/range custom dimensions can be created by anyone with Explore access.

📖 Docs: [Roles & permissions](https://docs.lightdash.com/references/roles)
